File: game/Assets/Scripts/GameLogic/Classes/Monster/MonsterGroup.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class MonsterGroup
{
    public string id;
    public List<Monster> monsters;
    public Vector2 position;

    public MonsterGroup(string id, Vector2 position)
    {
        this.position = position;
        this.id = id;
        this.monsters = new List<Monster>();
    }
}

File: server/src/Schema/MonsterGroup.ts
import Mongoose from "mongoose";
import Position from "../BusinessClasses/RelationalObject/Position";

export interface IMonsterGroup extends Mongoose.Document {
  monsters: Mongoose.Types.ObjectId[];
  position: { x: number; y: number };
  volatile: boolean;
}

export const MonsterGroupSchema = new Mongoose.Schema({
  monsters: [{ type: Mongoose.Schema.Types.ObjectId, ref: "Monster", default: [] }],
  position: { x: Number, y: Number },
  volatile: { type: Boolean, default: true }
});

const MonsterGroup = Mongoose.model<IMonsterGroup>("MonsterGroup", MonsterGroupSchema);
export default MonsterGroup;
